mono/utils/mono-threads.h

```diff
@@ -61,7 +61,7 @@ typedef gsize mono_thread_start_return_t;
 #define MONO_NATIVE_THREAD_ID_TO_UINT(tid) (gsize)(tid)
 #define MONO_UINT_TO_NATIVE_THREAD_ID(tid) (MonoNativeThreadId)(gsize)(tid)
 
-typedef gsize (*MonoThreadStart)(gpointer);
+typedef gulong (*MonoThreadStart)(gpointer);
```

Comment:
This is essentially nop for non-windows (though 32bit could use guint), but is I believe more correct for Windows, where threads always return a 32bit unsigned long.